.github/workflows/test.yml

```diff
@@ -0,0 +1,44 @@
+name: Twilio OAI Generator Test Workflow
+on:
+  push:
+    branches: ['*']
+  pull_request:
+    branches: [main]
+  schedule:
+    - cron: '0 15 * * 1-5'
+  workflow_dispatch:
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout twilio-oai-generator
+        uses: actions/checkout@v2
+
+      - name: Set up Java 11
+        uses: actions/setup-java@v2
+        with:
+          distribution: 'zulu'
+          java-version: '11'
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_AUTH_TOKEN }}
+
+      - name: Run tests
+        run: make test-docker
+
+      - name: Slack notify on failure
+        if: ${{ failure() && github.ref == 'refs/heads/main' && github.event_name != 'pull_request' }}
+        uses: rtCamp/action-slack-notify@v2
+        env:
+          SLACK_COLOR: ${{ job.status }}
+          SLACK_ICON_EMOJI: ':github:'
+          SLACK_MESSAGE: ${{ format('Build [#{2}]({0}/{1}/actions/runs/{2}) in {3} failed.', env.GITHUB_SERVER_URL, env.GITHUB_REPOSITORY, env.GITHUB_RUN_ID, github.repository) }}
```

Comment:
What's the difference between `env.GITHUB_REPOSITORY` and `github.repository`?

Reply:
I don't think there is a difference. I did reference this [github documentation](https://docs.github.com/en/actions/learn-github-actions/environment-variables#default-environment-variables) for this part. Note the blue box at the bottom of the section that tells you how to build the link for the workflow run's URL.